include/lib/map.h

```diff
@@ -4,24 +4,39 @@
 
 #include "stdlib/stdtypes.h"
 
-/** Data structure for a generic map */
+/** Structures for a generic map. */
+struct entry {
+    /** The key associated with the entry. */
+    void *key;
+    /** The value associated with the entry. */
+    void *value;
+};
+
 struct map {
-    //TODO
+    /** Table which holds all entries. */
+    struct entry *table;
+    /** Amount of space used. */
+    size_t used;
+    /** Maximum space to allocate for entries */
+    size_t size;
 };
 
-/** Initializes a new hashtable */
-void map_init(struct map *m);
+/** Initializes a new hashmap. Returns `-1` if not allocated. */
+int map_init(struct map *m, size_t size);
```

Comment:
I'm not sure what `size` is supposed to be here. Initializing the map should not take any extra parameters. You should decide what the default capacity for the map is and have it set to that constant instead of it being an input during initialization.